R/PlainTextParam.R

```diff
@@ -249,37 +252,40 @@ setMethod("loadResults", signature(object = "Spectra",
               s
           })
 
-
-
-# Notes: This and the Spectra method will be moved to it's respective package
 #' @rdname PlainTextParam
 setMethod("storeResults", signature(object = "MsBackendMzR",
                                     param = "PlainTextParam"),
           function(object, param) {
               dir.create(path = param@path,
                          recursive = TRUE,
-                         showWarnings = TRUE)
+                         showWarnings = FALSE)
               object <-  dropNaSpectraVariables(object)
-              write.table(object@spectraData,
-                          file = file.path(param@path, "backend_data.txt"),
-                          sep = "\t", quote = FALSE)
+              fl <- file.path(param@path, "backend_data.txt")
+              if (file.exists(fl))
+                  warning("Overwriting already present 'backend_data.txt' file")
+              writeLines(paste0("# ", class(object)[1L]), con = fl)
+              suppressWarnings(
+                  write.table(object@spectraData,
+                              file = fl, sep = "\t", quote = FALSE,
+                              append = TRUE, row.names = FALSE))
 })
 
 #' @rdname PlainTextParam
 setMethod("loadResults", signature(object = "MsBackendMzR",
                                     param = "PlainTextParam"),
           function(object, param, spectraPath = character()) {
               b <- MsBackendMzR()
-              data <- read.table(file = file.path(param@path,
-                                                  "backend_data.txt"),
-                                 sep = "\t", header = TRUE)
+              fl <- file.path(param@path, "backend_data.txt")
+              if (!file.exists(fl))
+                  stop("No 'backend_data.txt' file found in the provided path.")
+              data <- read.table(file = fl, sep = "\t", header = TRUE)
               rownames(data) <- NULL
               data <- DataFrame(data)
               b@spectraData <- data
-              if (length(spectraPath) > 0){
-                  old <- MsCoreUtils::common_path(dataStorage(b))
-                  if (nchar(old) > 0)
-                      old <- paste0(old, "/")
+              if (length(spectraPath) > 0) {
+                  old <- common_path(dataStorage(b))
+                  ## if (nchar(old) > 0)
```

Comment:
not sure if we need this - I would skip since `/` and `//` work both, so better to be sure that there is a `/` before the file name.

Reply:
mmm I need to test it, because I thought the result  of common_path never resulted with a `/` at the end of the path, so i thought that was necessary

Reply:
After checking, isn't that how the function behave ? and we definitely need to add this `/` ?

Reply:
how I understood it (can be wrong), but yes, the result from `common_path()` never ends in a `"/"`. so, assume we have `"/some/path/some/file.txt"`, then the common path would be `"/some/path/some" and assuming the user defines `spectraPath = "/new/path"`. We would like to replace to get `"/new/path/file.txt"`. If we always add a `"/"` at the `old`, `old` would be in this case `"/some/path/some/"` and the `sub()` call will replace `"/some/path/some/"` with `"/new/path"` - and we get `"/new/pathfile.txt"`. that's why I would not append the trailing `"/"` to the pattern we want to replace. 

Also, `"/new/path/file.txt"` and `"/new/path//file.txt"` both work - so, if the user would provide `spectraPath = "/new/path/"` (i.e. with a trailing `"/"`) it would/should still work.

Reply:
ooooh  of course  ! No that make complete sense, but ye I think that the less we have `if` statement the better so i will remove it.